### ISSUE TITLE
Need help regarding args.cmd when initializing a terminus

### ISSUE THREAD

**Issue by JW9506** [randy3k/Terminus#113]
`    {
    "caption": "venv testing",
    "command": "terminus_open",
    "args": {
        "cmd": ["source", "venv/bin/activate"],
        "title": "venv",
    }
    }, `

The above is my attempt to make terminus automatically execute "source venv/bin/activate", which is cool to have because I have venv inside the project folder. However, when I do so the window created doesn't function.

How do I work around this?

**Comment by randy3k** [comment]
Try `cmd: ["bash", "-c", "source venv/bin/activate"]`

**Comment by JW9506** [comment]
> Try `cmd: ["bash", "-c", "source venv/bin/activate"]`

It looks like it should work but the window popped up then disappeared immediately.

![Kapture 2019-05-04 at 16 21 47](https://user-images.githubusercontent.com/34848995/57184394-ea7a4500-6e88-11e9-8c7f-6e0ce563b1a0.gif)

**Comment by randy3k** [comment]
How about `cmd: ["bash", "-c", "source venv/bin/activate && exec bash -i -l"]`

**Comment by JW9506** [comment]
> How about `cmd: ["bash", "-c", "source venv/bin/activate && exec bash -i -l"]`
> 
> Edit: just tried, it doesn’t work

Yeah looks promising but doesn't work.

**Comment by randy3k** [comment]
One solution I found is to use the `—init-file` argument, but it will require you to create an alternate example rc file rather than `.bash_profile`

**Comment by JW9506** [comment]
> One solution I found is to use the `—init-file` argument, but it will require you to create an alternate rc file rather than `.bash_profile`

How does it work? Can you make some code I can attach to `.bash_profile` to solve this?

**Comment by randy3k** [comment]
Create the file `.bash_venv` in your project directory with the following content
```
source ~/.bash_profile
source activate venv/bin/activate
```

Then you could use `cmd: ["bash", "—init-file", ".bash_venv"]`

**Comment by randy3k** [comment]
It should be `--init-file` instead of `—init-file`. The darn auto correction.

**Comment by JW9506** [comment]
![Kapture 2019-05-04 at 18 42 34](https://user-images.githubusercontent.com/34848995/57185590-82cdf500-6e9c-11e9-9b66-ad976eb671bb.gif)

Tried this but then this came out..

**Comment by randy3k** [comment]
To be honest, I am on a mobile and didn’t test that. Let me try again later when I get back home.

**Comment by JW9506** [comment]
Okay~~

> To be honest, I am on a mobile and didn’t test that. Let me try again later when I get back home.

Okay~~

**Comment by JW9506** [comment]
> To be honest, I am on a mobile and didn’t test that. Let me try again later when I get back home.

Tried `cmd: ["bash", "-i", "-l", "--init-file", ".bash_venv"]`
Error msg:
![image](https://user-images.githubusercontent.com/34848995/57185634-59619900-6e9d-11e9-968b-e10af69aeb78.png)

**Comment by randy3k** [comment]
I tried many non-invasive methods but so far don't succeed. Here is a slight invasive method (requires editing ~/.bash_profile).

Appending the following in `~/.bash_profile`
```
if [ $TERM_PROGRAM == "Terminus-Sublime" ]; then
    if [ -z $TERMINUS_ACTIVATE_VENV ] && [ -f venv/bin/activate ] ; then
        source venv/bin/activate
    fi
fi
```

and consider the following setting
```
{ 
   "caption": "venv testing", 
   "command": "terminus_open", 
   "args": { 
			"cmd":
			[
				"bash", "-i", "-l"
			],
			"env": {
				"TERMINUS_ACTIVATE_VENV": "1"
			},
			"name": "VENV"
    } 
},
```

Basically, we use the environment variable `TERMINUS_ACTIVATE_VENV` to signal the activation.

**Comment by randy3k** [comment]
Okay, I finally was able to come up with something working.

```
			"cmd":
			[
				"bash", "-c", "exec bash --init-file <(echo '. ~/.bash_profile; source venv/bin/activate')"
			]
```

**Comment by JW9506** [comment]
Okay, tried these steps, but nothing happened. Please take a look again lol



![Kapture 2019-05-04 at 23 02 00](https://user-images.githubusercontent.com/34848995/57187679-c6d2f100-6ec0-11e9-90dd-cca11ef7d0ef.gif)

**Comment by JW9506** [comment]
Okay, worked like a charm. Thank you very much~~

**Comment by randy3k** [comment]
The command should look like this
```
{ 
   "caption": "venv testing", 
   "command": "terminus_open", 
   "args": { 
			"cmd":
			[
				"bash", "-c", "exec bash --init-file <(echo '. ~/.bash_profile; source venv/bin/activate')"
			]
}
```

We don't need the `TERMINUS_ACTIVATE_VENV` and editing `.bash_profile` anymore.
(Note, you will need to open the `venv testing` command in project window).
